docs/Upgrade.md

```diff
@@ -2121,42 +2123,32 @@ If you relied on the `handleSubmitWithRedirect` to change the redirection:
 ```diff
 const MyForm = () => {
     return (
-        <Create>
-            <FormWithRedirect
+-        <Create>
++        <Create redirect="show">
+-            <FormWithRedirect
++            <Form>
 -                render={({ handleSubmitWithRedirect, ...formProps }) => (
-+                render={({ handleSubmit, ...formProps }) => (
-+                    <form onSubmit={handleSubmit}>
                          <TextInput source="name" />
 -                        <MySaveButton handleSubmitWithRedirect={handleSubmitWithRedirect}>
 +                        <MySaveButton />
-+                    </form>
-                )}
-            />
+-                )}
+-            />
++            </Form>
         </Create>
     );
 };
 
-import { useSaveContext, useRedirect } from 'react-admin';
-import { useForm } from 'react-final-form';
 -const MySaveButton = ({ handleSubmitWithRedirect }) => (
 +const MySaveButton = () => {
-+    const { save } = useSaveContext();
-+    const form = useForm(); 
-+    const redirect = useRedirect();
-+    const handleClick = (event) => {
-+        event.preventDefault(); // Prevent the default form submission
-+        const values = form.getState().values;
-+        save(values, {
-+            onSuccess: (data) => redirect('show', '/posts', data.id)
-+        })
-+    };
     return (
 -        <button onClick={() => handleSubmitWithRedirect('show')}>Save</button>
-+        <button type="button" onClick={handleClick}>Save</button>
++        <button type="submit">Save</button>
```

Comment:
I don't understand this change. This SaveButton has the defautl behavior - it's not customized

Reply:
This example was about showing how to use `handleSubmitWithRedirect` to redirect to the **show** page after submission.

So in the new example I encouraged to make the following change:
```diff
-        <Create>
+        <Create redirect="show">
```
And then you don't need to customize the Save button anymore.

What is troubling you? Did you miss the `redirect="show"` I added, or is it because the `redirect` prop is more limited than what you could do with `handleSubmitWithRedirect` and you would rather have me add a more complete example?